credentials/oauth/oauth.go

```diff
@@ -133,10 +134,13 @@ func NewComputeEngine() credentials.PerRPCCredentials {
 // serviceAccount represents PerRPCCredentials via JWT signing key.
 type serviceAccount struct {
 	config *jwt.Config
+	once   sync.Once
+	ts     oauth2.TokenSource
 }
 
-func (s serviceAccount) GetRequestMetadata(ctx context.Context, uri ...string) (map[string]string, error) {
-	token, err := s.config.TokenSource(ctx).Token()
+func (s *serviceAccount) GetRequestMetadata(ctx context.Context, uri ...string) (map[string]string, error) {
+	s.once.Do(func() { s.ts = s.config.TokenSource(ctx) })
```

Comment:
This means that `ctx` will continue to be used even after it has been cancelled by the caller. Doesn't seem bueno.